src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -158,8 +158,10 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 interfaceDocumentId,
                 cancellationToken);
 
-            var solutionWithFormattedInterfaceDocument = GetSolutionWithFormattedInterfaceDocument(unformattedInterfaceDocument, cancellationToken);
-
+            var syntaxFactsService = refactoringResult.DocumentToExtractFrom.GetLanguageService<ISyntaxFactsService>();
+            var fileBanner = syntaxFactsService.GetFileBanner(refactoringResult.DocumentToExtractFrom.GetSyntaxRootSynchronously(cancellationToken));
```

Comment:
and I still think you need to move this to GetUnformattedInterfaceDocument and pass in refactoringResult.DocumentToExtractFrom rather than refactoringResult.DocumentToExtractFrom.Folders

I dont think "fileBanner" needs to leak out and passed around. you can put that to the tree when tree is generated.